tests/ops/test_channel_ops.py

```diff
@@ -152,6 +159,24 @@ def test_p_arbitrary(self, p, tol):
         expected_K1 = np.sqrt(p) * Z
         assert np.allclose(op(p, wires=0).kraus_matrices[1], expected_K1, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("angle", np.linspace(0, 2 * np.pi, 7))
+    def test_grad_phaseflip(self, angle, tol):
+        """Test that analytical gradient is computed correctly for different states. Channel
+        grad recipes are independent of channel parameter"""
+
+        dev = qml.device("default.mixed", wires=1)
+        prob = 0.5
+
+        @qml.qnode(dev)
+        def circuit(p):
+            qml.RX(angle, wires=0)
+            qml.PhaseFlip(p, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        gradient = np.squeeze(qml.grad(circuit)(prob))
+        assert gradient == circuit(1) - circuit(0)
+        assert np.allclose(gradient, 0.0)
```

Comment:
How come the gradient is 0? If we change the state prep (e.g., to RY), does the gradient become non-zero?

Reply:
Yes, gradient would generally be non-zero for different states. I actually think it's a good check that we get a zero gradient in this case